misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/completion/FunctionCompletionItemBuilder.java

```diff
@@ -0,0 +1,151 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.debugadapter.completion;
+
+import io.ballerina.compiler.api.symbols.ArrayTypeSymbol;
+import io.ballerina.compiler.api.symbols.FunctionSymbol;
+import io.ballerina.compiler.api.symbols.FunctionTypeSymbol;
+import io.ballerina.compiler.api.symbols.ParameterSymbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.ballerinalang.debugadapter.completion.context.CompletionContext;
+import org.ballerinalang.debugadapter.completion.util.CommonUtil;
+import org.eclipse.lsp4j.debug.CompletionItem;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class is being used to build function type completion item.
+ *
+ * @since 2201.1.0
+ */
+public final class FunctionCompletionItemBuilder {
+
+    private static final String PARENTHESIS = "()";
+    private static final String OPEN_PARENTHESIS = "(";
+    private static final String CLOSE_PARENTHESIS = ")";
+
+    private FunctionCompletionItemBuilder() {
+    }
+
+    /**
+     * Creates and returns a completion item.
+     *
+     * @param functionSymbol BSymbol
+     * @param context        Debug completion Context
+     * @return {@link CompletionItem}
+     */
+    public static CompletionItem build(FunctionSymbol functionSymbol, CompletionContext context) {
+        CompletionItem item = new CompletionItem();
+        if (functionSymbol != null && functionSymbol.getName().isPresent()) {
```

Comment:
If the `functionSymbol` is `null` why do we return an empty completion item? We should avoid reaching to this point in that case.